### DISCUSSION TITLE
How can we use PhysicalFile.TrySetAttributeAsync(string name, object value)

### DISCUSSION THREAD

**Discussion by gmarginet** [VsixCommunity/Community.VisualStudio.Toolkit#383] [Q&A] [answered]
Hello,

I want to use the PhysicalFile.TrySetAttributeAsync(...) to store custom settings for a certain file.

I already used Project.TrySetAttributeAsync(...) to store custom settings for a certain project.
The custom setting is then stored in the project file (after calling Project.SaveAsync()).

I would expected I could use PhysicalFile.TrySetAttributeAsync(...) in the same way, and that it would be stored in the project file the phisical file belongs to, but that doesn't seems to work?

What i'm trying to achieve?
I'm creating a source code generator and I want to give my user the ability to define what parts need to be generated in the code behind file. So the custom setting "GeneratePart1" equals true or false and should be stored together with the source file the generator uses. The ultimate goal would be that this new custom property can be changed from the file properties tool window. To do that I tried to create a Property extender and provider, but changing the value also doesn't get stored.

So if somebody has an other idea to to this?

Kind regards,
Gert

**Comment by reduckted** [comment, marked as answer]
I just tried setting a custom property and noticed that it was able to get the `VSHPROPID_BrowseObject` property value, but because the custom attribute I was trying to set did not have a property descriptor, it was not able to set the property via the property descriptor. It _should_ have been falling back to writing the attribute to the project file, but that was skipped.

This should be fixed by #384.

**Reply by gmarginet** [reply]
I tested your code change via a local build of the toolkit and this fixes the issue.
Any idea when the PR will be aproved and when it will be available in a nuget version?
